contracts/Polymorph.sol

```diff
@@ -58,23 +57,21 @@ contract Polymorph is IPolymorph, ERC721PresetMinterPauserAutoId, BMath, Reentra
     }
 
     function _beforeTokenTransfer(address from, address to, uint256 tokenId) internal virtual override(ERC721PresetMinterPauserAutoId) {
-        if(whitelistAddresses[from] == true || whitelistAddresses[to] == true) {
-            ERC721PresetMinterPauserAutoId._beforeTokenTransfer(from, to, tokenId);
-            emit MarketplaceTransfer(tokenId, from, to);
-        } else {
-            uint256 oldGene = _genes[tokenId];
-            _genes[tokenId] = geneGenerator.random();
-            ERC721PresetMinterPauserAutoId._beforeTokenTransfer(from, to, tokenId);
-            emit TokenMorphed(tokenId, oldGene, _genes[tokenId]);
-        }
+        ERC721PresetMinterPauserAutoId._beforeTokenTransfer(from, to, tokenId);
     }
 
     function mint() public override payable nonReentrant {
         _tokenIdTracker.increment();
+
+        uint256 tokenId = _tokenIdTracker.current();
```

Comment:
Use the same variable in the _mint function instead of calling `current()` again